weave-js/src/components/PagePanelComponents/Home/Browse2/tableStats.ts

```diff
@@ -0,0 +1,122 @@
+import {GridColumnVisibilityModel} from '@mui/x-data-grid-pro';
+import {useState} from 'react';
+
+type ValueType =
+  | 'undefined'
+  | 'null'
+  | 'boolean'
+  | 'string'
+  | 'number'
+  | 'other';
+
+const determineType = (value: any): ValueType => {
+  if (value === null) {
+    return 'null';
+  }
+  if (value === undefined) {
+    return 'undefined';
+  }
+  if (typeof value === 'boolean') {
+    return 'boolean';
+  }
+  if (typeof value === 'string') {
+    return 'string';
+  }
+  if (typeof value === 'number') {
+    return 'number';
+  }
+  return 'other';
+};
+
+type ColumnStats = {
+  valueCount: number;
+  // TODO: Would make code more complex but could only store counts for types seen
+  typeCounts: Record<ValueType, number>;
+};
+type TableStats = {
+  rowCount: number;
+  column: Record<string, ColumnStats>;
+};
+
+export const computeTableStats = (table: Array<Record<string, any>>) => {
```

Comment:
Flagging as a full pass over data (which we do in many spots). Depending on the number of calls in view, this could get expensive.

Reply:
Yes - I timed it at a small number of ms for various projects we have so far but definitely something to keep in mind, especially if we start adding another calculations like min/max that might be pretty nice for highlighting cells.